examples/gno.land/p/demo/accesscontrol/accesscontrol.gno

```diff
@@ -0,0 +1,111 @@
+// Package accesscontrol provides a library for managing roles and
+// permissions within Gno. It allows for the creation,
+// assignment, and management of roles with specific admin privileges,
+// ensuring that only authorized accounts can perform certain actions.
+// This package utilizes an AVL tree data structure to efficiently track
+// and manage role holders.
+//
+// Example Usage:
+//
+//  import "gno.land/p/demo/accesscontrol"
+//
+//  Create a new role with a specific admin.
+//  adminRole := std.Address("admin-address")
+//  role := accesscontrol.NewRoleData("ExampleRole", adminRole)
+//
+//  Check if an account has a specific role.
+//  account := std.Address("user-address")
+//  hasRole := role.HasRole(account)
+//
+//  Grant a role to a specific account.
+//  role.GrantRole(account)
+//
+//  Revoke a role from a specific account.
+//  role.RevokeRole(account)
+//
+//  Renounce a role with caller confirmation.
+//  role.RenounceRole(std.GetOrigCaller())
+//
+//  Change the admin role for a specific role.
+//  newAdmin := std.Address("new-admin-address")
+//  role.SetRoleAdmin(newAdmin)
+
+package accesscontrol // import "gno.land/p/demo/accesscontrol"
+
+import (
+	"std"
+	"gno.land/p/demo/avl"
+)
+
+// RoleData struct to store role information
+type RoleData struct {
+	Name      string
+	Holder    *avl.Tree // -> std.Address -> bool
+	AdminRole std.Address
+}
+
+// NewRoleData creates a new instance of RoleData
+func NewRoleData(name string, adminRole std.Address) *RoleData {
+	return &RoleData{
+		Name:      name,
+		Holder:    avl.NewTree(),
+		AdminRole: adminRole,
+	}
+}
+
+// Method to check if the caller has the admin role
+func (roleData *RoleData) OnlyAdmin() {
+	caller := std.GetOrigCaller()
+	if roleData.AdminRole != caller {
+		panic("accesscontrol: caller does not have the admin role")
```

Comment:
I replaced the method name and returned errors instead of panics.
- https://github.com/gnolang/gno/pull/2307/commits/91a21eab5b2d0668ad27d7adc7e16087f6e416c7

I used to use panic in realm, but it's true that it's better to return errors in packages. Thanks for this clarification